test/actions/katello/content_view_version/export_test.rb

```diff
@@ -13,12 +12,12 @@ def setup
       @content_view = @repo.content_view
       @content_view_version = @repo.content_view_version
       @destination_server = 'dream-destination'
-      FileUtils.mkdir_p(Setting['pulpcore_export_destination'])
```

Comment:
Since this failed due to a permissions issue, you could override the setting value to a Tmpdir -> https://ruby-doc.org/stdlib-2.5.1/libdoc/tmpdir/rdoc/Dir.html or `Rails.root.join('tmp')`

The tmpdir will be writable by jenkins, and I'd like to see that approach here because I don't want to get bitten by this again - although I see you've added a workaround.

Reply:
One may say stub everything you aren't testing but I would say look what happens when you do that! ;)

Reply:
@jturel yeah, I was thinking about that but I decided to just improve the stub (and make sure it's unstubbed after the tests). I think the `unstub` will prevent the future biting, and feels more correct than actually altering the filesystem on the test machine.  thoughts?

Reply:
It's not world-ending but I prefer creating our own prescribed dir because:

- someone might come and delete that line in the teardown if they see the tests pass (I think Jenkins will even be green if that test runs toward the end)
- Stubbing such a high level construct (File) tends to be brittle in general. *any* code that gets executed could affect the stub - so a gem update might cause it to fail for example.

TLDR - just writing our own file will probably prevent us from ever having to look at this test again for the same or similar reason we had to this time around and for me that's a huge plus

Reply:
If you do want to keep the stub then I think the line you put in the teardown must have a comment about why it's critical cause removing it is not as innocuous as it seems

Reply:
(just for future Github readers) We ended up going with the previous approach of not stubbing `File.directory` at all.